blobfile/_azure.py

```diff
@@ -1180,6 +1187,9 @@ def _upload_chunk(self, chunk: memoryview, finalize: bool) -> None:
             data = chunk[start:end]
             self._md5.update(data)
             block_md5 = hashlib.md5(data)
+            headers = {"Content-MD5": base64.b64encode(block_md5.digest()).decode("utf8")}
+            if self._version is not None:
+                headers["If-Match"] = self._version

```

Comment:
does put block actually take this header? if so, what does it do?

Reply:
Yeah, sorry, that didn't make sense. Removed.